Realm/RLMRealm.mm

```diff
@@ -702,22 +707,21 @@ - (void)deleteObject:(RLMObject *)object {
     RLMDeleteObjectFromRealm(object, self);
 }
 
-- (void)deleteObjects:(id)array {
-    if ([array respondsToSelector:@selector(realm)] && [array respondsToSelector:@selector(deleteObjectsFromRealm)]) {
-        if (self != (RLMRealm *)[array realm]) {
+- (void)deleteObjects:(id<NSFastEnumeration>)objects {
+    NSObject *collection = (NSObject *)objects;
+    if ([collection respondsToSelector:@selector(realm)]
+        && [collection respondsToSelector:@selector(deleteObjectsFromRealm)]) {
+        if (self != (RLMRealm *)[collection performSelector:@selector(realm)]) {
```

Comment:
Just cast to `id` rather than using `performSelector`. It's less verbose and there's only a functional difference if there's multiple possible return types for that selector (with the cast to `id` it'll fail to compile; with `performSelector` it'll produce a warning and leak memory if you ignore that).